code/game/machinery/electrolyzer.dm

```diff
@@ -0,0 +1,181 @@
+#define ELECTROLYZER_MODE_STANDBY	"standby"
+#define ELECTROLYZER_MODE_WORKING	"working"
+
+/obj/machinery/electrolyzer
+	anchored = FALSE
+	density = TRUE
+	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_WIRES_IF_OPEN
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "electrolyzer-off"
+	name = "space electrolyzer"
+	desc = "Thanks to the fast and dynamic response of our electrolyzers, on-site hydrogen production is guaranteed. Warranty void if used by clowns"
+	max_integrity = 250
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 80, "acid" = 10)
+	circuit = /obj/item/circuitboard/machine/electrolyzer
+	ui_x = 400
+	ui_y = 305
+
+	///used to check if there is a cell in the machine
+	var/obj/item/stock_parts/cell/cell = null
+	///check if the machine is on or off
+	var/on = FALSE
+	///check what mode the machine should be (WORKING, STANDBY)
+	var/mode = ELECTROLYZER_MODE_STANDBY
+	///Increase the amount of moles worked on, changed by upgrading the laser tier
+	var/workingPower = 1
+	///Decrease the amount of power usage, changed by upgrading the capacitor tier
+	var/efficiency = 0.5
+
+/obj/machinery/electrolyzer/get_cell()
+	return cell
+
+/obj/machinery/electrolyzer/Initialize()
+	. = ..()
+	cell = new cell(src)
```

Comment:
`cell = new cell(src)` and `cell = new (src)` is the same.

This will give you a free cell on `Initialize()` same as before, the behavior is unchanged.

Why are you building a new cell here in `Initialize()`? Is it required?

Reply:
Yes you are right, i had set the cell as null, so that's the reason i thought it worked even with that part of code in there. But i discovered that this choice made the electrolyzer runtime when built. Now i removed the `cell = new cell(src)` from there and it works, without generating a cell and without runtiming